### PR TITLE
Internal Node Protocol

### PR DESCRIPTION
**Pull request by chathamabate**
Hey guys,

This is a draft of the proto file I was thinking we could use for cluster node to cluster node communication.

Let me know what y'all think,
Chatham

### CODE REVIEW

File: protos/node.proto [hunk 1]
```diff
@@ -0,0 +1,134 @@
+
+syntax = "proto3";
+
+import "./head.proto"
```
Comment (josh8551021) [inline protos/node.proto:4]: Probably better to use fully-qualified names for imports.

File: protos/node.proto [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    
+    // ID of the parent directory.
+    UUID parent_dir_id = 3;
+
+    // ID and name of the new directory.
+    UUID new_dir_id = 4;
+    string new_dir_name = 5;
+
+message MakeDirectoryCSResponse {
+    // Empty for now.
```
Comment (josh8551021) [inline protos/node.proto:41]: We could probably add a dir_id field (or more) here to ensure that the response knows that the correct directory was made.

File: protos/node.proto [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    string new_file_name = 4;
+
+    // Contents of file being added...
+    // If not provided, an empty file
+    // will be created.
+    bytes contents = 5;
+}
+
+message AddNewFileCSResponse {
+    // Empty for now.
```
Comment (josh8551021) [inline protos/node.proto:62]: Same as above, and for all other response messages.

**Review by Landmaster — APPROVED** [review]
The stuff you have so far looks good to me.

**Pull request merged**